src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/LoggingExtensions.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Microsoft.Extensions.Logging;
+using System;
+using System.Text;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions
+{
+    internal static class LoggingExtensions
```

Comment:
I agree with the idea that these don't really belong as extension methods. However, using a static class means we have to pass around ILogger alot which is a little cumbersome. Was it considered to have a class that takes optionally takes ILogger and wraps it to avoid passing it every time?

Reply:
I hadn't really considered it, does that mean that class would get stored in AzureAppConfigurationProvider?

Reply:
> does that mean that class would get stored in AzureAppConfigurationProvider?

It would be it's own class that `AzureAppConfigurationProvider` instantiates.

```
class Logger
{
    public Logger();

    public Logger(ILogger logger);

    public LogInformation(string message);

    public LogWarning(string message, Exception e);
}
```

Usage
```
_logger = logger != null ?
    new Logger(logger) :
    new Logger();
```

Reply:
Ok thanks that's what I was expecting. Don't see any problem with this at the moment.